easybuild/framework/easyblock.py

```diff
@@ -331,10 +331,12 @@ def get_checksum_for(self, checksums, filename=None, index=None):
                 return checksums[index]
             else:
                 return None
+        elif isinstance(checksums, dict):
+            return checksums.get(filename)
```

Comment:
@edmondac Please ensure proper error reporting if `filename` is not a key in `checksums`